File: android-app/app/src/main/java/com/borehole/management/data/local/entity/JobEntity.kt
package com.borehole.management.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "jobs")
data class JobEntity(
    @PrimaryKey
    val id: String,
    val name: String,
    val clientName: String,
    val siteName: String,
    val latitude: Double,
    val longitude: Double,
    val status: String,
    val contactPerson: String? = null,
    val contactPhone: String? = null,
    val accessNotes: String? = null,
    val priority: String? = null,
    val budgetUsd: Double? = null,
    val assignedSurveyorId: String? = null,
    val assignedDrillerId: String? = null,
    val assignedAt: Date? = null,
    val surveyCompletedAt: Date? = null,
    val createdAt: Date,
    val updatedAt: Date,
    val lastSyncedAt: Date? = null
)

File: backend/src/jobs/job.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn, OneToMany } from 'typeorm';
import { User } from '../users/user.entity';

export enum JobStatus {
  CREATED = 'created',
  ASSIGNED = 'assigned',
  SURVEYED = 'surveyed',
  DRILLING = 'drilling',
  COMPLETED = 'completed',
}

@Entity('jobs')
export class Job {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  name: string;

  @Column()
  clientName: string;

  @Column()
  siteName: string;

  @Column('decimal', { precision: 10, scale: 8 })
  latitude: number;

  @Column('decimal', { precision: 11, scale: 8 })
  longitude: number;

  @Column({ type: 'enum', enum: JobStatus, default: JobStatus.CREATED })
  status: JobStatus;

  @Column({ nullable: true })
  contactPerson: string;

  @Column({ nullable: true })
  contactPhone: string;

  @Column({ nullable: true })
  accessNotes: string;

  @Column({ nullable: true })
  priority: string;

  @Column('decimal', { precision: 10, scale: 2, nullable: true })
  budgetUsd: number;

  @ManyToOne(() => User, { nullable: true })
  @JoinColumn({ name: 'assignedSurveyorId' })
  assignedSurveyor: User;

  @Column({ nullable: true })
  assignedSurveyorId: string;

  @ManyToOne(() => User, { nullable: true })
  @JoinColumn({ name: 'assignedDrillerId' })
  assignedDriller: User;

  @Column({ nullable: true })
  assignedDrillerId: string;

  @Column({ nullable: true })
  assignedAt: Date;

  @Column({ nullable: true })
  surveyCompletedAt: Date;

  @Column({ nullable: true })
  drillingStartedAt: Date;

  @Column({ nullable: true })
  drillingCompletedAt: Date;

  @Column('json', { nullable: true })
  drillingResults: {
    finalDepth?: number;
    waterYield?: number;
    isSuccessful?: boolean;
    completedAt?: Date;
    completedBy?: string;
  };

  @Column('json', { nullable: true })
  surveyResults: {
    recommendedDepth?: number;
    soilType?: string;
    waterTableDepth?: number;
    notes?: string;
  };

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}